assets/app/views/directives/events.html

```diff
@@ -1,37 +1,80 @@
-<table class="table table-bordered table-condensed table-mobile table-hover events-table">
-  <thead>
-    <tr>
-      <th>Time</th>
-      <th class="hidden-xs hidden-sm hidden-md sr-only">Severity</th>
-      <th>Reason</th>
-      <th>Message</th>
-    </tr>
-  </thead>
-  <tbody ng-if="(filteredEvents | hashSize) === 0">
-    <tr>
-      <td><em>{{emptyMessage}}</em></td>
-      <td class="hidden-xs hidden-sm hidden-md">&nbsp;</td>
-      <td class="hidden-xs">&nbsp;</td>
-      <td class="hidden-xs">&nbsp;</td>
-    </tr>
-  </tbody>
-  <tbody ng-repeat="event in filteredEvents | orderBy:'-lastTimestamp'">
-    <tr>
-      <td data-title="Time" class="nowrap">{{event.lastTimestamp | date:'mediumTime'}}</td>
-      <td data-title="Severity" class="hidden-xs hidden-sm hidden-md text-center severity-icon-td">
-        <span class="sr-only">{{event.type}}</span>
-        <span class="pficon pficon-warning-triangle-o" ng-show="event.type === 'Warning'" aria-hidden="true" data-toggle="tooltip" data-placement="right" data-original-title="Warning"></span></td>
-      <td data-title="Reason" class="event-time">
-        {{event.reason}}&nbsp;
-        <span class="hidden-lg pficon pficon-warning-triangle-o" ng-show="event.type === 'Warning'" aria-hidden="true" data-toggle="tooltip" data-placement="right" data-original-title="Warning"></span>
-      </td>
-      <td data-title="Message" class="word-break">
-        {{event.message}}
-        <span class="text-muted small" ng-if="event.count > 1">
-          ({{event.count}} times in the last
-            {{event.firstTimestamp | duration:event.lastTimestamp:true}})
-        </span>
-      </td>
-    </tr>
-  </tbody>
-</table>
+<div ng-if="!events">
+  Loading...
+</div>
+<div ng-if="events">
+  <div class="table-toolbar form-inline">
+    <div class="form-group filter-controls">
+      <label for="events-filter" class="sr-only">Filter</label>
+      <input type="search"
+             placeholder="Filter by keyword"
+             class="form-control"
+             id="events-filter"
+             ng-model="filter.text">
+    </div>
+    <div class="vertical-divider"></div>
+    <!-- TODO: Change to pf-sort in angular-patternfly 3.0 -->
+    <div pf-simple-sort config="sortConfig" class="sort-controls"></div>
+  </div>
+  <table class="table table-bordered table-condensed table-mobile table-hover events-table">
+    <thead>
+      <tr>
+        <th>Time</th>
+        <!-- Only show Name and Kind columns if not passed to the directive. -->
+        <th ng-if="!resourceKind || !resourceName">
+          <span class="hidden-xs-inline visible-sm-inline visible-md-inline hidden-lg-inline">Resource</span>
+          <span class="visible-lg-inline">Name</span>
```

Comment:
We still are. I changed the header to "Resource" from "Kind and Name". I can change it back if that's confusing.